imagepuller/internal/service/verify.go

```diff
@@ -84,52 +85,32 @@ func (s *ImagePullerService) getAndVerifyImage(ctx context.Context, log *slog.Lo
 	return remoteImg, nil
 }
 
-func (s *ImagePullerService) storeAndVerifyLayers(log *slog.Logger, remoteImg gcr.Image) (string, error) {
+func (s *ImagePullerService) storeAndVerifyLayers(log *slog.Logger, remoteImg gcr.Image) error {
 	layers, err := remoteImg.Layers()
 	if err != nil {
-		return "", fmt.Errorf("obtaining the image layers: %w", err)
+		return fmt.Errorf("obtaining the image layers: %w", err)
 	}
 
 	manifest, err := remoteImg.Manifest()
 	if err != nil {
-		return "", fmt.Errorf("obtaining image manifest: %w", err)
+		return fmt.Errorf("obtaining image manifest: %w", err)
 	}
 
-	previousLayer := ""
-	for idx, layer := range layers {
-		rc, err := layer.Compressed()
-		if err != nil {
-			return "", fmt.Errorf("reading layer %d: %w", idx, err)
-		}
+	// TODO(burgerdev): pass a context here?
```

Comment:
I might be missing something here, but it seems like we should quite easily be able to pass in the context from `PullImage` here?